ledger/state_test.go

```diff
@@ -0,0 +1,464 @@
+// Copyright 2025 Blink Labs Software
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ledger
+
+import (
+	"testing"
+
+	"github.com/blinklabs-io/dingo/chain"
+	"github.com/blinklabs-io/dingo/database"
+	"github.com/blinklabs-io/dingo/database/models"
+	ocommon "github.com/blinklabs-io/gouroboros/protocol/common"
+)
+
+// TestGetBlock_Success tests that GetBlock returns a valid block when found
+func TestGetBlock_Success(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create test block
+	testBlock := models.Block{
+		Hash:     []byte{0x01, 0x02, 0x03, 0x04},
+		Slot:     12345,
+		Number:   100,
+		PrevHash: []byte{0x00, 0x00, 0x00, 0x00},
+		Cbor:     []byte{0x82, 0x00, 0x81, 0x00}, // minimal valid CBOR
+		Type:     5,                               // Byron
+	}
+
+	// Add block to chain
+	primaryChain := cm.PrimaryChain()
+	if err := primaryChain.AddBlock(testBlock, nil); err != nil {
```

Comment:
@coderabbitai This should be a "github.com/blinklabs-io/gouroboros/ledger/common".Block not a models.Block struct. You can use `Decode()` on models.Block to get a ledger.Block suitable for use here.